File: Threshold.cpp
#include <iostream>
#include <fstream>
#include <stdlib.h>
using namespace std;

class Threshold{
	private:
		int** thr_Ary;
		int numRow;
		int numCol;
		int minVal;
		int maxVal;
		int thresholdVal;
		ifstream in;
	    ofstream out;
		string file1, file2;
		
	public:
		Threshold(string input){
			file1 = input;
			
			in.open(file1.c_str());
			int value;	
			if(in>>value) numRow = value;
		    if(in>>value) numCol = value;
			if(in>>value) minVal = value;
			if(in>>value) maxVal = value;
			
			thr_Ary = new int*[numRow];
			for(int i=0; i<numRow; i++) thr_Ary[i] = new int[numCol];
			
			for(int i=0; i<numRow; i++){
				for(int j=0 ; j<numCol ; j++){
					thr_Ary[i][j] = 0;
				}
			}
		}
		
		~Threshold(){
			for(int i=0 ; i<numRow; i++) delete[] thr_Ary[i];
			delete[] thr_Ary;
		}
		
		void computeThreshold(){
			string num;
			cout<<"Please enter a number: ";
			cin>>num;
			thresholdVal = atoi(num.c_str());
			file2 = "thr_" + num +".txt";
			out.open(file2.c_str());
			out<<numRow<<" "<<numCol<<" "<<0<<" "<<1<<endl;
			
			int input;
			for(int i=0; i<numRow; i++){
				for(int j=0; j<numCol; j++){
					if(in>>input){
						if(input >= thresholdVal) {
							thr_Ary[i][j] = 1;
							out<< thr_Ary[i][j]<<" ";
						}
						else {
							thr_Ary[i][j] = 0;
							out<< thr_Ary[i][j]<<" ";
						}
					}
				}
				out<<endl;
			}
			in.close();
			out.close();
		}	
};

int main(int argc, char* argv[]){
	if(argc < 2){
		cout << "usage : " << argv[0] << " <input file>\n";
		return -1;
	}//check the command-line arguments
	
	string f1;
	f1 = argv[1];
	Threshold h(f1);
	h.computeThreshold();
	return 0;
}

File: CS381 PROJECT1.2 Threshold/src/Threshold.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Threshold {
	private int[][] thr_Ary;
	private int numRow;
	private int numCol;
	private int minVal;
	private int maxVal;
	int thresholdVal;
    private File file1, file2;
    private Scanner scan;
    private java.io.PrintStream output;
    
    public Threshold(String input){
    	file1 = new File(input);	
    	
    	try {
			scan = new Scanner(file1);
			if(scan.hasNextInt()) numRow = scan.nextInt();
			if(scan.hasNextInt()) numCol = scan.nextInt();
			if(scan.hasNextInt()) minVal = scan.nextInt();
			if(scan.hasNextInt()) maxVal = scan.nextInt();
			
			thr_Ary = new int[numRow][numCol];
			for(int i=0; i<numRow; i++){
				for(int j=0 ; j<numCol ; j++){
					thr_Ary[i][j] = 0;
				}
			}
			
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		}
    }
    
    void computeThreshold(){
    	Scanner reader = new Scanner(System.in); 
    	System.out.println("Please enter a number: ");
    	thresholdVal = reader.nextInt();
    	String output_filename = "thr_" + thresholdVal +".txt";
    	file2 = new File(output_filename);
    	
    	try {
			output = new java.io.PrintStream(file2);
			output.println(numRow+" "+numCol+" "+0+" "+1);
			int input;
			for(int i=0; i<numRow; i++){
				for(int j=0; j<numCol; j++){
					if(scan.hasNextInt()){
						input = scan.nextInt();
						if(input >= thresholdVal) {
							thr_Ary[i][j] = 1;
							output.print(thr_Ary[i][j]+" ");
						}
						else {
							thr_Ary[i][j] = 0;
							output.print(thr_Ary[i][j]+" ");
						}
					}
				}
				output.println();
			}
			output.close();
			scan.close();
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		}
    }

}
